src/ergw_api.erl

```diff
@@ -45,6 +47,11 @@ delete_contexts(all) ->
     lists:foreach(fun(Context) ->
         gtp_context:trigger_delete_context(Context)
     end, contexts(all));
+delete_contexts(#{} = Criteria) ->
+    lists:foreach(fun(Pid) ->
+        {_, State} = sys:get_state(Pid),
```

Comment:
No,

Don't access process state like state. Either filter through the context_reg or when that doesn't have enough information, use a dedicated API on the context to get that information.